src/server/pkg/worker/api_server.go

```diff
@@ -490,6 +490,9 @@ func HashDatum(pipelineInfo *pps.PipelineInfo, data []*Input) (string, error) {
 		hash.Write(datum.FileInfo.Hash)
 	}
 
+	// We set env to nil because if env contains more than one elements,
+	// since it's a map, the output of Marshal() can be non-deterministic.
+	pipelineInfo.Transform.Env = nil
```

Comment:
That's what I thought originally too, but turns out that the env vars are set as part of the `ReplicaSet` which is done by the PPS master when the workers are created.  So the `Env` field isn't directly used by the workers.